subsys/bluetooth/host/conn.c

```diff
@@ -1545,14 +1545,15 @@ void bt_conn_unref(struct bt_conn *conn)
 	conn_handle = conn->handle;
 
 	old = atomic_dec(&conn->ref);
-	/* Prevent from accessing connection object */
-	conn = NULL;
-	deallocated = (atomic_get(&old) == 1);
 
 	LOG_DBG("handle %u ref %ld -> %ld", conn_handle, old, (old - 1));
 
 	__ASSERT(old > 0, "Conn reference counter is 0");
 
+	/* Prevent from accessing connection object */
+	conn = NULL;
```

Comment:
The comment is rather misleading. This is a local variable, and the function doesn't access it anyway after this point.

Reply:
I think the purpose of the comment is specifically to ensure that the local variable cannot accidentally be used. I suppose this is similar to https://doc.rust-lang.org/std/mem/fn.drop.html but not a pattern that we often see in C

Reply:
> The comment is rather misleading. This is a local variable, and the function doesn't access it anyway after this point.

I agree. I kept this comment assuming that the intent was to "Prevent *those who will read this in ten years* from accessing connection object *from here until the end of the function*".

IMO we should either:
- move a reworded comment and the `conn = NULL` statement to the `if (deallocated)` block (disposing the connection pointer regardless of its reference counter is also confusing)
- or remove both the comment and the `conn = NULL` statement